### DISCUSSION TITLE
segments/precision

### DISCUSSION THREAD

**Discussion by hrgdavor** [jscad/OpenJSCAD.org#899] [Ideas]
When doing design precision of curves can have big impact on preview performance, and in that phase of working with the model I would prefer performance over precision of the preview.

**Comment by platypii** [comment]
If you've every used d3, they allow you to set attributes on elements either as a constant, or using a function. For example in D3:

```js
element.attr("width", 3)
element.attr("width", (d) => 10 * d)
```

It would be cool if we could do something similar in JSCAD. Where you could set segments either as a constant, or as a function of some "quality level":

```js
cylinder({ segments: 6 }) // constant hex nut
cylinder({ segments: (quality) => 100 * quality }) // dynamic quality setting depending on what the renderer wants to do
```

This would allow cool things like fast on screen rendering at low quality, for example while adjusting parameters. But then can be rendered at higher quality for high quality rendering for 3D printing, or for screenshots.

I would also vote that the _default_ be the dynamic setting, since it represents a "true" cylinder in some sense, just a question how accurately you want to approximate it.
